peer/doc.go

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2015 The btcsuite developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+/*
+Package peer provides a common base for creating and managing bitcoin network
+peers for fully validating nodes, Simplified Payment Verification (SPV) nodes,
+proxies, etc. It includes basic protocol exchanges like version negotiation,
+responding to pings etc.
+
+Inbound peers accept a connection and respond to the version message to begin
+version negotiation.
+
+Outbound peers dial a remote address and use the connection to push the initial
+version message.
+
+Both peers accept a configuration to customize user agent, service flag,
```

Comment:
I'd add a "such as" here since I don't believe this list covers everything that is configurable.